app/controllers/reactions_controller.rb

```diff
@@ -133,6 +139,27 @@ def rate_article(reaction)
                       rating: current_user.experience_level)
   end
 
+  def clear_article_reactions(id, type, mod, category)
+    reactions = if category == "thumbsup"
+                  Reaction.where(reactable_id: id, reactable_type: type, user: mod).where.not(category: category)
+                elsif category.in?(NEG_ARTICLE_REACTIONS)
+                  Reaction.where(reactable_id: id, reactable_type: type, user: mod, category: "thumbsup")
+                end
+    reactions.each { |reaction| destroy_reaction(reaction) }
```

Comment:
Two questions (see my previous question about understanding the purpose):

1. can we use `reactions.find_each` to only load a few reactions at the time
2. can all of this be executed asynchronously?

Reply:
re: Question 1 - I will do so 👍 
re: Question 2 - probably so! 😄 I think this goal could be part of a future PR though

Reply:
TBH I think this should only be destroying two reactions at most, since we're looking for either a thumbs down + vomit reaction from the mod that is reacting.

Reply:
Cool! I thought we were removing all reactions :D